### PR TITLE
Update documentation for release v1.5.0

### PR DESCRIPTION
**Pull request by vshampor**

### CODE REVIEW

**Comment by vshampor** [comment]
Please add or correct anything I've missed.

**Review by ljaljushkin — CHANGES_REQUESTED** [review]
please merge my changes https://github.com/vshampor/nncf_pytorch/pull/2

**Review by ljaljushkin — APPROVED** [review]

**Review by asenina — APPROVED** [review]

File: nncf/config_schema.py [hunk 1]
```diff
@@ -678,7 +678,14 @@ def with_attributes(schema: Dict, **kwargs) -> Dict:
         # This is required for better user feedback, since holistic schema validation is uninformative
         # if there is an error in one of the compression configs.
         "compression": make_object_or_array_of_objects_schema(BASIC_COMPRESSION_ALGO_SCHEMA),
-        "target_device": TARGET_DEVICE_SCHEMA,
+        "target_device": with_attributes(TARGET_DEVICE_SCHEMA,
+                                         description="The target device, the specificity of which will be taken into "
+                                                     "account while compressing in order to obtain the best "
+                                                     "performance for this type of device. The default 'ANY' means "
+                                                     "compatible quantization supported by any HW. The parameter takes "
+                                                     "values from the set ('CPU', 'GPU', 'VPU', 'ANY', 'NONE'). If you "
```
[the remark this answers is not in the document]
Comment (vshampor) [reply nncf/config_schema.py:686]: Done

File: ReleaseNotes.md [hunk 2]
```diff
@@ -7,6 +7,23 @@ samples distributed with the code.  The samples demonstrate the usage of compres
 public models and datasets for three different use cases: Image Classification, Object Detection,
 and Semantic Segmentation.
 
+## New in Release 1.5:
+- Switched to using the propagation-based mode for quantizer setup by default. Compared to the previous default, pattern-based mode, the propagation-based mode better ensures that all the inputs to operations that can be quantized on a given type of hardware are quantized in accordance with what this hardware allows. Default target hardware is CPU - adjustable via `"target_device"` option in the NNCF config.
```
[the remark this answers is not in the document]
Comment (vshampor) [reply ReleaseNotes.md]: Done

File: nncf/config_schema.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vshampor) [reply nncf/config_schema.py]: Done

**Pull request merged**